review10/README.md

```diff
@@ -0,0 +1,148 @@
+
+#### Bearer access token
+
+Bearer access token is a type of access token that is used to authenticate API requests. The client sends HTTP requests with the Authorization header that contains the word Bearer followed by a space and a token. The server uses the token to authenticate the user.
+
+A token can be issued after successful login using Basic authentication or SSO. The token can be used to access other services without the need to provide a username and password again. It's useful for user authentication, so he doesn't need to write a username and password for every request.
+
+However, the problem with the token invalidation comes. If the token is stolen, the attacker can use it until it expires. To mitigate this problem, the token can be short-lived, and the client can use a refresh token to get a new token.
+
+#### Mutual TLS authentication
+
+Mutual TLS authentication is a type of authentication that uses client certificates to authenticate the client. The client sends HTTP requests with the client certificate. The server uses the client certificate to authenticate the client.
+
+#### Single sign-on (SSO)
+
+Single sign-on (SSO) is a type of authentication that allows users to log in once and access multiple systems without re-authenticating. The client sends HTTP requests with the Authorization header that contains the word Bearer followed by a space and a token. The server uses the token to authenticate the user against the identity provider.
+
+### HTTPS
+
+When the application is running in a production environment, it's important to use HTTPS to secure the communication between the client and the server. HTTPS is a secure version of HTTP that uses SSL/TLS to encrypt the data transmitted between the client and the server. It provides a secure channel for the client and the server to communicate over the internet. **Always use HTTPS in a production environment.** However, we can use HTTP in our development environment.
+
+### Examples for Basic authentication
+
+```java
+@Path("/api/public")
+public class PublicResource {
+
+    @GET
+    @PermitAll // allow all access
+    @Produces(MediaType.TEXT_PLAIN)
+    public String publicResource() {
+        return "public";
+   }
+   
+    @GET
+    @RolesAllowed("user") // allow only users with role "user" to access this resource
+    @Path("/me")
+    public String me(@Context SecurityContext securityContext) { // inject the security context
+      return securityContext.getUserPrincipal().getName();
+    }
+
+    @GET
+    @RolesAllowed("admin") // allow only users with role "admin" to access this resource 
+    @Produces(MediaType.TEXT_PLAIN)
+    public String adminResource() {
+      return "admin";
+    }
+}
+```
+
+## State of the project
+
+- Added `security-jpa-reactive` extension to and `baggage-service`.
+- In `baggage-service` added `User` entity for managing users for basic authentication. Check `User` entity in `baggage-service` for more details.
```

Comment:
nick picking but it's kind of strange to keep users in baggage-service when you have passenger or even flight. In the best case, that would be a separate user-service.

Reply:
Users in this context are other services